frappe/database/query.py

```diff
@@ -422,36 +456,73 @@ def function_objects_from_list(self, fields):
 
 	def remove_string_functions(self, fields, function_objects):
 		"""Remove string functions from fields which have already been converted to function objects"""
+
+		def _remove_string_aliasing(function, fields: list | str):
+			if function.alias:
+				to_replace = " as " + function.alias.casefold()
+				if to_replace in fields:
+					fields = fields.replace(to_replace, "")
+				elif " as " + f"`{function.alias.casefold()}" in fields:
+					fields = fields.replace(" as " + f"`{function.alias.casefold()}`", "")
+			return fields
+
 		for function in function_objects:
 			if isinstance(fields, str):
-				if function.alias:
-					fields = fields.replace(" as " + function.alias.casefold(), "")
-				fields = BRACKETS_PATTERN.sub("", fields.replace(function.name.casefold(), ""))
+				fields = _remove_string_aliasing(function, fields)
+				fields = BRACKETS_PATTERN.sub("", fields.casefold().replace(function.name.casefold(), ""))
+				# Converting back to capitalized doctype names.
+				if "tab" in fields:
+					fields = TABLE_PATTERN.sub(
+						lambda p: p.group(0)[:3] + p.group(0)[3].upper() + p.group(0)[3 + 1 :], fields
+					)
 				# Check if only comma is left in fields after stripping functions.
 				if "," in fields and (len(fields.strip()) == 1):
 					fields = ""
 			else:
 				updated_fields = []
 				for field in fields:
 					if isinstance(field, str):
-						if function.alias:
-							field = field.replace(" as " + function.alias.casefold(), "")
-						field = (
-							BRACKETS_PATTERN.sub("", field).strip().casefold().replace(function.name.casefold(), "")
+						field = _remove_string_aliasing(function, field)
+						substituted_string = (
+							BRACKETS_PATTERN.sub("", field).strip().casefold()
+							if "`" not in field
+							else BRACKETS_PATTERN.sub("", field).strip()
 						)
-						updated_fields.append(field)
-
+						# This is done to avoid casefold of table name.
+						if substituted_string.casefold() == function.name.casefold():
+							replaced_string = substituted_string.casefold().replace(function.name.casefold(), "")
+						else:
+							replaced_string = substituted_string.replace(function.name.casefold(), "")
+						updated_fields.append(replaced_string)
 					fields = [field for field in updated_fields if field]
```

Comment:
same here, the *whole* iterable is getting changed while iterating on it

Reply:
I think I'm still making a new container and `replacing` values in it and I think I need to replace the string functions passed in as fields to replace them as pypika function objects

Reply:
😟 I removed iterable from loop 😢